pkg/lvmd/cmd/root.go

```diff
@@ -97,31 +99,53 @@ func subMain(ctx context.Context) error {
 	proto.RegisterVGServiceServer(grpcServer, vgService)
 	proto.RegisterLVServiceServer(grpcServer, lvmd.NewLVService(dcm, ocm, notifier))
 	grpc_health_v1.RegisterHealthServer(grpcServer, lvmd.NewHealthService())
-	well.Go(func(ctx context.Context) error {
-		return grpcServer.Serve(lis)
-	})
-	well.Go(func(ctx context.Context) error {
-		<-ctx.Done()
+
+	var wg sync.WaitGroup
+	errCh := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := grpcServer.Serve(lis); err != nil {
+			errCh <- err
+		} else {
+			errCh <- nil
+		}
+		cancel()
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		childCtx, stop := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
+		defer stop()
+
+		<-childCtx.Done()
+
+		cancel()
 		grpcServer.GracefulStop()
```

Comment:
Could you consider a smart way to this logic by merging this block with the `select` block of the ticker?

Reply:
I fixed my code. Could you review it again? The changes I made is [here](https://github.com/topolvm/topolvm/compare/711bbff5c42498048d55e86c3e5914b5228b546b..e9a0ff19d8ea63d5509718f337859973c901f153).